airflow/kubernetes/pod_launcher.py

```diff
@@ -143,17 +148,90 @@ def base_container_is_running(self, pod):
         wait=tenacity.wait_exponential(),
         reraise=True
     )
-    def read_pod_logs(self, pod):
+    def _read_pod_log_chunk(self, pod: Pod, last_line: bytes) -> HTTPResponse:
+        # The CoreV1Api doesn't support since_time even though the API does, so we must use
+        # since_seconds. Add 15 seconds of buffer just in case of NTP woes
+        if last_line:
+            # Strip fractional part because strptime doesn't support nanosecond parsing
+            timestamp = last_line.split(b" ", 1)[0]
+            last_chunk_dt = dt.strptime(timestamp.split(b".", 1)[0].decode("utf-8"),
+                                        "%Y-%m-%dT%H:%M:%S")
+            since_time = last_chunk_dt
+        else:
+            since_time = dt.utcfromtimestamp(0)
+        since_seconds = math.ceil((dt.utcnow() - since_time).total_seconds() + 15)
+        resp = self._client.read_namespaced_pod_log(
+            name=pod.name,
+            namespace=pod.namespace,
+            container='base',
+            follow=False,
+            since_seconds=since_seconds,
+            timestamps=True,
+            _preload_content=False
+        )
 
-        try:
-            return self._client.read_namespaced_pod_log(
-                name=pod.name,
-                namespace=pod.namespace,
-                container='base',
-                follow=True,
-                tail_lines=10,
-                _preload_content=False
+        # If we've already read a chunk, skip until we find a matching line
+        # Just in case since_seconds doesn't get everything we want, keep the previous lines in a buffer
+        buffered_lines = []
+        skipping_lines = True
+        for line in resp:
+            if skipping_lines:
+                if line == last_line:
+                    self.log.debug("Found duplicate line. Stopping log skipping")
+                    buffered_lines = []
+                    skipping_lines = False
+                    for buffered_line in buffered_lines:
+                        yield buffered_line
+                else:
+                    buffered_lines.append(line)
```

Comment:
will `resp` ever contain lines before `last_line`?

I may be misreading but:

While `skipping_lines` is true and we haven't hit the `last_line` yet, then we'll be buffering potentially older lines that have already been yielded, but aren't the `last_line`. And then when we _do_ hit `last_line` we yield all of the buffered lines and then the rest of the lines in the response.

Reply:
That for loop It doesn't do anything right now since it's set to [] immediately above. Not sure how it got there